### DISCUSSION TITLE
MikTex+TeXstudio编译失败

### DISCUSSION THREAD

**Discussion by darkblue-76** [sjtug/SJTUThesis#1064] [提问] [answered]
没有动模版，按照readme设置了编译器，但是texstudio编译失败，没有PDF产生，显示
main.tex:24: Paragraph ended before \environment SJTU@CJK@FTS  was complete.
<to be read again> 
                   \par 
l.24 ^^I
[main.log](https://github.com/user-attachments/files/17615595/main.log)
win10，MiKTeX 24.4，TeXstudio
请问该如何解决？

**Comment by darkblue-76** [comment]
gb7714-2015宏包在MiKTeX里已经装了

**Comment by hushidong** [comment]
编译选择用biber了么？

获取Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by darkblue-76** [reply]
<img width="448" alt="1730728668166" src="https://github.com/user-attachments/assets/21840cb6-4efa-4076-85ed-bcbace01a4ca">
这个吗，也没用

**Comment by AlexaraWu** [comment, marked as answer]
目前已经确认是 `array` 包 v2.6g 版本更新造成的问题。

可以在模板根目录放一个旧版本的 `array.sty` 作为临时的解决办法。

**Reply by darkblue-76** [reply]
加了之后能生成pdf了，但是pdf是损坏文件打不开，  
xdvipdfmx: Command for 'xdvipdfmx' gave return code 256
这是新的log
[main.log](https://github.com/user-attachments/files/17626456/main.log)

**Reply by LogCreative** [reply]
xdvipdfmx 的问题在日志文件中是显示不出来的，只有命令行能看到。

可以考虑先清理一下之前的临时文件再进行编译？

**Reply by darkblue-76** [reply]
清理了再运行还是一样的，命令行显示是这样
`Clean files . . .
Initial Win CP for (console input, console output, system): (CP65001, CP65001, CP936)
I changed them all to CP936
Rc files read:
  .latexmkrc

Compile . . .
Initial Win CP for (console input, console output, system): (CP65001, CP65001, CP936)
I changed them all to CP936
Rc files read:
  .latexmkrc
Latexmk: Run number 1 of rule 'xelatex'
This is XeTeX, Version 3.141592653-2.6-0.999996 (MiKTeX 24.4) (preloaded format=xelatex.fmt)
 restricted \write18 enabled.
entering extended mode
SyncTeX written on main.synctex.gz.
Latexmk: Getting log file 'main.log'
Latexmk: Run number 1 of rule 'biber main'
Biber warning: [663] Biber.pm:131> WARN - legacy year field '1957/1990' in entry 'LSC1957' is not an integer - this will probably not sort properly.
Latexmk: Run number 2 of rule 'xelatex'
This is XeTeX, Version 3.141592653-2.6-0.999996 (MiKTeX 24.4) (preloaded format=xelatex.fmt)
 restricted \write18 enabled.
entering extended mode
SyncTeX written on main.synctex.gz.
Latexmk: Getting log file 'main.log'
Latexmk: Run number 3 of rule 'xelatex'
This is XeTeX, Version 3.141592653-2.6-0.999996 (MiKTeX 24.4) (preloaded format=xelatex.fmt)
 restricted \write18 enabled.
entering extended mode
SyncTeX written on main.synctex.gz.
Latexmk: Getting log file 'main.log'
Latexmk: Run number 4 of rule 'xelatex'
This is XeTeX, Version 3.141592653-2.6-0.999996 (MiKTeX 24.4) (preloaded format=xelatex.fmt)
 restricted \write18 enabled.
entering extended mode
SyncTeX written on main.synctex.gz.
Latexmk: Getting log file 'main.log'
Latexmk: Run number 1 of rule 'xdvipdfmx'
Collected error summary (may duplicate other messages):
  xdvipdfmx: Command for 'xdvipdfmx' gave return code 256


Error! Please check the 'main.log' for more details . . .`
[main.log](https://github.com/user-attachments/files/17626857/main.log)

**Reply by AlexaraWu** [reply]
试试手动执行 `xdvipdfmx` 是什么结果？

```bash
xdvipdfmx -E -o "main.pdf" "main.xdv"
```

**Reply by darkblue-76** [reply]
<img width="876" alt="1730789164866" src="https://github.com/user-attachments/assets/f2e641f4-162e-4402-a922-72f467fd003f">

文件夹里是有main.pdf和main.xdv的

**Reply by LogCreative** [reply]
确认一下那个图片是否存在（以及后缀名是什么），如果存在的话，可以考虑把路径改为英文路径再试试？

**Reply by darkblue-76** [reply]
可以！我没动模版文件夹，图片应该是自带的，我把路径名改了之后就OK了！

**Reply by darkblue-76** [reply]
谢谢各位大佬们的热心帮助！感激不尽！

**Reply by JohnJackal** [reply]
> 目前已经确认是 `array` 包 v2.6g 版本更新造成的问题。
> 
> 可以在模板根目录放一个旧版本的 `array.sty` 作为临时的解决办法。

大佬好，请问最近的一个 `array` 包的可用版本号是多少啊，我今天重装windows后将texlive从2022更新到了2024，也出现了与题主一样的问题。

**Reply by JohnJackal** [reply]
OK没事了，我把array-2020-02-10.sty放进sjtuthesis根目录后编译通过了。感谢！
